### PR TITLE
allow vulkan validation layers in release builds if explicity asked for

### PR DESCRIPTION
**Pull request by jordo**

### CODE REVIEW

File: main/main.cpp [hunk 1]
```diff
@@ -698,9 +698,9 @@ Error Main::setup(const char *execpath, int argc, char *argv[], bool p_second_ph
 		} else if (I->get() == "-w" || I->get() == "--windowed") { // force windowed window
 
 			init_windowed = true;
-#ifdef DEBUG_ENABLED
 		} else if (I->get() == "--vk-layers") {
 			Engine::singleton->use_validation_layers = true;
+#ifdef DEBUG_ENABLED
```
Comment (akien-mga) [inline main/main.cpp:703]: It should also be changed further up for the help output.

**Comment by bruvzg** [comment]
Note: This flags only controls loading of `VK_LAYER_KHRONOS_validation` and debug prints via `VK_EXT_debug_utils`.

All layers set by `VK_LAYER_PATH` and `VK_INSTANCE_LAYERS` environment variables or layers with `.json` in the `$HOME/.local/share/vulkan/implicit_layer.d` / `/usr/local/share/vulkan/implicit_layer.d` / Windows registry will be loaded anyway (e.g. RenderDoc capture, Steam Overlay UI).

**Comment by jordo** [comment]
This flag eventually gets passed into the vulkan_context.cpp, and ultimately controls whether or not this runs:
`if (use_validation_layers) {
		_create_validation_layers();
	}` which also enables validation in that class.  Without this flag, all that is ignored.  It's compiled into the binary regardless of release or debug.  So why not allow it in release as well?

Yes, and there's definitely a lot of environment stuff to setup as well, but this also needs to be passed into the vulkan context to work properly at the application level.

**Comment by bruvzg** [comment]
Probably this line - https://github.com/godotengine/godot/blob/master/drivers/vulkan/vulkan_context.cpp#L305 should have `enabled_layers` instead of `instance_validation_layers`, and this one - https://github.com/godotengine/godot/blob/master/drivers/vulkan/vulkan_context.h#L123 removed (variable not used).

**Comment by akien-mga** [comment]
> Probably this line - https://github.com/godotengine/godot/blob/master/drivers/vulkan/vulkan_context.cpp#L305 should have `enabled_layers` instead of `instance_validation_layers`, and this one - https://github.com/godotengine/godot/blob/master/drivers/vulkan/vulkan_context.h#L123 removed (variable not used).

Seems correct indeed.

@jordo Could you squash the commits into one? If you want to also include the bugfix suggested by @bruvzg, feel free to.

**Comment by bruvzg** [comment]
> but this also needs to be passed into the vulkan context to work properly at the application level.

I do not think so, implicitly enabled layers should be loaded without setting list in the `VkInstanceCreateInfo` (that's what `_create_validation_layers()` should do, but not doing due to aforementioned bug).

> Loader implementations may provide mechanisms outside the Vulkan API for enabling specific layers. Layers enabled through such a mechanism are implicitly enabled, while layers enabled by including the layer name in the ppEnabledLayerNames member of VkInstanceCreateInfo are explicitly enabled. Implicitly enabled layers are loaded before explicitly enabled layers, such that implicitly enabled layers are closer to the application, and explicitly enabled layers are closer to the driver. Except where otherwise specified, implicitly enabled and explicitly enabled layers differ only in the way they are enabled, and the order in which they are loaded. Explicitly enabling a layer that is implicitly enabled results in this layer being loaded as an implicitly enabled layer; it has no additional effect.

From https://www.khronos.org/registry/vulkan/specs/1.2-extensions/html/vkspec.html#extendingvulkan-layers

**Comment by jordo** [comment]
@akien-mga  OK I'll add @bruvzg fix and squash everything into one.

**Comment by jordo** [comment]
@akien-mga squashed.

**Review by bruvzg — APPROVED** [review]

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks! And congrats for your first Godot contribution :tada:

**Pull request merged**